### ISSUE TITLE
Stratified sampling for dataframe splitting

### ISSUE THREAD

**Issue by codeheart09** [javascriptdata/danfojs#355]
Hello everyone,

I'm working on a project that needs stratified sampling of the dataset so it can have a more balanced test set.
More on the subject: https://en.wikipedia.org/wiki/Stratified_sampling 

I implemented a solution using Danfo.js for that purpose and, if you think it is a good idea, I can open a PR with that as a splitting tool.
Its parallel in scikit learn: https://scikit-learn.org/stable/modules/generated/sklearn.model_selection.StratifiedShuffleSplit.html

If you think this would make sense for the project, just let me know. :)

**Comment by risenW** [comment]
Hello @codeheart09 Thanks for the suggestion, and yes, we would love this, but we are currently working on a [Scikit.js library](https://github.com/javascriptdata/scikit.js), and this will fit right into it. Specifically, in this folder: https://github.com/javascriptdata/scikit.js/tree/main/src/model_selection

I can transfer this issue there, and then assign you to it. What do you say?

**Comment by codeheart09** [comment]
Hey @risenW, sure! You can transfer it and I get it from there!
